### PR TITLE
Return the expected exit code if sky_shell sees a compilation error during script loading

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth @a-siva

**Comment by abarth** [comment]
LGTM

File: runtime/dart_controller.cc [hunk 1]
```diff
@@ -129,8 +129,9 @@ tonic::DartErrorHandleType DartController::RunFromSource(
   Dart_Handle result = loader.LoadScript(main);
   LogIfError(result);
```
Comment (a-siva) [inline runtime/dart_controller.cc]: I think we may need a check of result here as loader.LoadScript calls Dart_LoadScript which does a top level compile
of the loaded script.

**Comment by Hixie** [comment]
There's some code in flutter_tools that verifies the exit code of sky_shell when running tests which you could make cleverer if it can recognise this exit code.

File: runtime/dart_controller.cc (hunk 1 above)
Comment (jason-simmons) [inline runtime/dart_controller.cc]: The GetErrorHandleType() check after this line should handle that case

**Pull request merged**